src/main/java/gift/domain/Category/CategoryService.java

```diff
@@ -1,56 +1,58 @@
+import org.springframework.transaction.annotation.Transactional;
 
-    @Service
-    public class CategoryService {
+@Service
+public class CategoryService {
 
-        private final JpaCategoryRepository categoryRepository;
+    private final gift.domain.category.JpaCategoryRepository categoryRepository;
 
-        public CategoryService(JpaCategoryRepository categoryRepository) {
-            this.categoryRepository = categoryRepository;
-        }
-
-        public List<Category> getCategories() {
-            List<Category> categories = categoryRepository.findAll();
-            return categories;
-        }
+    public CategoryService(gift.domain.category.JpaCategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
 
-        public void createCategory(CategoryDTO categoryDTO) {
-            if (categoryRepository.existsByName(categoryDTO.getName())) {
-                throw new BusinessException(HttpStatus.BAD_REQUEST, "동일한 이름의 카테고리 존재");
-            }
+    public List<Category> getCategories() {
+        List<Category> categories = categoryRepository.findAll();
+        return categories;
+    }
 
-            Category category = new Category(categoryDTO.getName(), categoryDTO.getDescription());
-            categoryRepository.save(category);
+    public void createCategory(CategoryDTO categoryDTO) {
+        if (categoryRepository.existsByName(categoryDTO.getName())) {
+            throw new CategoryDuplicateException(categoryDTO.getName());
         }
 
-        public void deleteCategory(Long id) {
-            if (categoryRepository.findById(id).isEmpty()) {
-                throw new BusinessException(HttpStatus.BAD_REQUEST, "해당 카테고리가 존재 X");
-            }
+        Category category = new Category(categoryDTO.getName(), categoryDTO.getDescription());
+        categoryRepository.save(category);
+    }
 
-            categoryRepository.deleteById(id);
+    public void deleteCategory(Long id) {
+        if (categoryRepository.findById(id).isEmpty()) {
+            throw new CategoryNotFoundException(id);
         }
 
-        @Transactional
-        public void updateCategory(Long id, CategoryDTO categoryDTO) {
-            Optional<Category> findCategory = categoryRepository.findById(id);
+        categoryRepository.deleteById(id);
+    }
 
-            if (findCategory.isEmpty()) {
-                throw new BusinessException(HttpStatus.BAD_REQUEST, "수정할 카테고리가 존재 X");
-            }
-            // 이름 중복 검사, 중복되면서 id 가 자신이 아닐 때
-            Optional<Category> compareCategory = categoryRepository.findByName(categoryDTO.getName());
-            if (compareCategory.isPresent() && compareCategory.get().getId() != id) {
-                throw new BusinessException(HttpStatus.BAD_REQUEST, "해당 이름의 카테고리 이미 존재");
-            }
+    @Transactional
+    public void updateCategory(Long id, CategoryDTO categoryDTO) {
+        Optional<Category> findCategory = categoryRepository.findById(id);
```

Comment:
이렇게 사용하면 Optional이 보다 바람직하게 사용될 것 같아요!
```suggestion
        Category findCategory = categoryRepository.findById(id)
            .orElseThrow(() -> new CategoryNotFoundException(id));
```

Reply:
수정했습니다! 감사합니다!